src/lib/features/scheduler/scheduler-service.ts

```diff
@@ -9,8 +10,14 @@ export class SchedulerService {
 
     private logger: Logger;
 
-    constructor(getLogger: LogProvider) {
+    private maintenanceService: MaintenanceService;
+
+    constructor(
+        getLogger: LogProvider,
+        maintenanceService: MaintenanceService,
+    ) {
         this.logger = getLogger('/services/scheduler-service.ts');
+        this.maintenanceService = maintenanceService;
```

Comment:
Maybe this should be a read model instead of exposing the entire maintenance service. I'd like it to happen in a follow up though PR if we go that route since I need this merged.